x/cron/keeper/keeper.go

```diff
@@ -89,7 +94,13 @@ func (k *Keeper) AddSchedule(ctx sdk.Context, name string, period uint64, msgs [
 		Period:            period,
 		Msgs:              msgs,
 		LastExecuteHeight: uint64(ctx.BlockHeight()), // let's execute newly added schedule on `now + period` block
+		ExecutionStage:    executionStage,
 	}
+
+	if _, ok := types.ExecutionStage_name[int32(executionStage)]; !ok {
+		schedule.ExecutionStage = types.ExecutionStage_EXECUTION_STAGE_END_BLOCKER
+	}
```

Comment:
discussed with @pr0n00gler — we don't need a default value for exec stage, and we better explicitly require the passed value to be in the list, otherwise return an error. Please add this kind of validation to the Validate method

Reply:
Done